starknet_devnet/blocks.py

```diff
@@ -99,18 +99,22 @@ async def generate(
         """
         block_number = self.get_number_of_blocks()
         timestamp = state.state.block_info.block_timestamp
-        signature = transaction.get_signature()
-
         if block_number == 0:
             parent_block_hash = 0
         else:
             last_block = self.__get_last_block()
             parent_block_hash = last_block.block_hash
 
-        if self.lite:
+        if is_empty_block:
+            transaction_receipts = ()
+            transactions = []
             block_hash = block_number
         else:
-            block_hash = await calculate_block_hash(
+            if self.lite:
```

Comment:
Couldn't nested ifs somehow be avoided? By having something like `if is_empty_block or self.lite`?

Reply:
how about this?